test cases/common/191 has link arg/meson.build

```diff
@@ -40,3 +40,7 @@ assert(l2.length() == 0, 'First supported did not return empty array.')
 assert(not cc.has_multi_link_arguments([isnt_arg, is_arg]), 'Arg that should be broken is not.')
 assert(cc.has_multi_link_arguments(is_arg), 'Arg that should have worked does not work.')
 assert(cc.has_multi_link_arguments([useless, is_arg]), 'Arg that should have worked does not work.')
+
+if cc.has_link_argument('-Wl,-z,nodelete')
+  assert(not cc.has_link_argument('-Wl,-z,nodelete42'), 'Did not detect wrong -z linker argument')
+endif
```

Comment:
Actually, don't we also want to ensure that this works for other compilers and linkers? Such as clang and msvc.

I think this test should be:

```meson
if cc.get_id() == 'gcc' # might be os-specific too
  assert(cc.has_link_argument('-Wl,-z,nodelete'), 'msg')
endif
assert(not cc.has_link_argument('-Wl,-z,nodelete42'), 'msg')
```

Reply:
... i.e., I don't see why the `assert(not ...)` is conditional if it's supposed to fail for all linkers.